src/types/token_stream.cpp

```diff
@@ -291,6 +294,39 @@ void TokenStream::determineNewlineSectors()
 		}
 		else if (!leftBrackets.empty() && (current == NEW_LINE || current == OR || current == AND))
 			leftBrackets.top()->setFlag(true);
+
+		// Here we count lines and double lines:
+		if (current == NEW_LINE)
+		{
+			++lineCounter;
+			if (wasLine)
+				++doubleLineCounter;
+			wasLine = true;
+		}
+		else
+			wasLine = false;
+	}
+	return 3 * doubleLineCounter > lineCounter;
```

Comment:
This means that when more than half of the newlines is "doubled", then all "doubled" newlines are made simple new line.

Reply:
This kind of comment would better fit directly in the source code :) Thanks.